src/domain/gift-cards/index.tsx

```diff
@@ -1,85 +1,77 @@
-import GiftCardTable from "../../components/templates/gift-card-table"
-import PageDescription from "../../components/atoms/page-description"
-import { useAdminGiftCards } from "medusa-react"
-import Spinner from "../../components/atoms/spinner"
+import Overview from "./overview"
 
-const Index = () => {
-  const [query, setQuery] = useState({
-    offset: 0,
-    limit: 20,
+const GiftCard = () => {
+  const { gift_cards: purchasedGiftCards } = useAdminGiftCards()
+  const { products: giftCards } = useAdminProducts({
+    is_giftcard: "true",
   })
+  const giftCard = giftCards?.[0]
 
-  const { gift_cards, isLoading } = useAdminGiftCards(query)
+  const updateGiftCard = useAdminUpdateProduct(giftCard?.id)
+  const deleteGiftCard = useAdminDeleteProduct(giftCard?.id)
 
-  const searchQuery = (q) => {
-    const baseUrl = qs.parseUrl(window.location.href).url
+  const toaster = useToaster()
 
-    const search = {
-      q,
-      offset: 0,
-      limit: 20,
+  const updateGCStatus = () => {
+    let status: ProductStatus = ProductStatus.PUBLISHED
+    if (giftCard?.status === "published") {
+      status = ProductStatus.DRAFT
     }
 
-    const prepared = qs.stringify(search, {
-      skipNull: true,
-      skipEmptyString: true,
-    })
-
-    window.history.replaceState(baseUrl, "", `?${prepared}`)
+    updateGiftCard.mutate(
+      { status },
+      {
+        onSuccess: () => toaster("Successfully updated Gift Card", "success"),
+        onError: (err) => toaster(getErrorMessage(err), "error"),
+      }
+    )
   }
 
-  const actionables = [
-    {
-      label: "Custom Gift Card",
-      onClick: () => console.log("create custom gift card"), // TODO
-      icon: <PlusIcon size={20} />,
-    },
-  ]
-
-  const { setOnSearch, onUnmount } = useContext(InterfaceContext)
-  useEffect(onUnmount, [])
-  useEffect(() => {
-    setOnSearch(searchQuery)
-  }, [])
+  const updateGC = (data) => {
+    updateGiftCard.mutate(
+      { ...data },
+      {
+        onSuccess: () => toaster("Successfully updated Gift Card", "success"),
+        onError: (err) => toaster(getErrorMessage(err), "error"),
+      }
+    )
+  }
 
-  return (
-    <div className="flex flex-col grow h-full">
-      <div className="w-full flex flex-col grow">
-        <PageDescription
-          title="Gift Cards"
-          subtitle="Manage the Gift cards for your Medusa Store"
-        />
-        <BodyCard
-          title="History"
-          subtitle="See the history of purchased gift cards"
-          actionables={actionables}
-        >
-          {isLoading || !gift_cards ? (
-            <Spinner size="large" />
-          ) : (
-            <GiftCardTable giftCards={gift_cards} />
-          )}
-        </BodyCard>
-      </div>
-    </div>
-  )
-}
+  const deleteGC = () => {
+    deleteGiftCard.mutate(undefined, {
+      onSuccess: () => navigate("/a/gift-cards"),
+    })
+  }
 
-const GiftCard = () => {
   return (
     <Router>
```

Comment:
Router should be full height to better align with the design
```suggestion
    <Router className="h-full">
```